xarray/core/common.py

```diff
@@ -938,6 +938,39 @@ def ones_like(other, dtype=None):
     return full_like(other, 1, dtype)
 
 
+def label_like(data, other):
+    """Return a new object labeled in the same way as the given object.
+
+    Parameters
+    ----------
+    data : array_like
+        Data to fill the new object. Must have same shape as other.
+    other : DataArray, or Variable
+        The reference object from which to get labels
+
+    Returns
+    -------
+    out : same as other
+        New object with the same shape, and properties as other.
+        Coords will be copied from other.
+    """
+    from .dataarray import DataArray
+    from .variable import Variable
+
+    if not (isinstance(other, DataArray) or isinstance(other, Variable)):
+        raise TypeError("Expected DataArray, or Variable")
```

Comment:
I think it could make logical sense to support coercing a dict of numpy arrays into an xarray.Dataset (as the `data_vars` argument). But I'm also happy to leave that for a later improvement.

Reply:
Yeah I wasn't sure of the best approach and I couldn't think of a time when I'd use it (I was mostly working with DataArrays) so I figured it would be ok to throw an error for now. But I am happy to add it if that is more normal.